source/mechanics/force_contribution_lib.cc

```diff
@@ -815,6 +816,128 @@ namespace fdl
+    const unsigned int                     index_f,
+    const double                           a_s,
+    const double                           b_s,
+    const double                           kappa_s,
+    const unsigned int                     index_s,
+    const double                           a_fs,
+    const double                           b_fs,
+    std::shared_ptr< FiberNetwork< dim, spacedim> >
+                                           fiber_network, 
+    const std::vector<types::material_id> &material_ids)
+    : ForceContribution<dim, spacedim, Number>(quad)
+    , a(a)
+    , b(b)
+    , a_f(a_f)
+    , b_f(b_f)
+    , kappa_f(kappa_f)
+    , index_f(index_f)
+    , a_s(a_s)
+    , b_s(b_s)
+    , kappa_s(kappa_s)
+    , index_s(index_s)
+    , a_fs(a_fs)
+    , b_fs(b_fs)
+    , fiber_network(fiber_network)
+    , material_ids(setup_ids(material_ids))
+  {}
+
+  template <int dim, int spacedim, typename Number>
+  MechanicsUpdateFlags
+  ModifiedHolzapfelOgdenStress<dim, spacedim, Number>::get_mechanics_update_flags()
+    const
+  {
+    return MechanicsUpdateFlags::update_FF |
+           MechanicsUpdateFlags::update_modified_first_invariant | 
+           MechanicsUpdateFlags::update_modified_first_invariant_dFF |
+           MechanicsUpdateFlags::update_right_cauchy_green;
+  }
+
+  template <int dim, int spacedim, typename Number>
+  bool
+  ModifiedHolzapfelOgdenStress<dim, spacedim, Number>::is_stress() const
+  {
+    return true;
+  }
+
+  template <int dim, int spacedim, typename Number>
+  void
+  ModifiedHolzapfelOgdenStress<dim, spacedim, Number>::compute_stress(
+    const double /*time*/,
+    const MechanicsValues<dim, spacedim>                              &m_values,
+    const typename Triangulation<dim, spacedim>::active_cell_iterator &cell,
+    ArrayView<Tensor<2, spacedim, Number>> &stresses) const
+  {
+    if (this->material_ids.size() > 0 &&
+        !std::binary_search(this->material_ids.begin(),
+                            this->material_ids.end(),
+                            cell->material_id()))
+      {
+        // the user specified a subset of material ids and we currently don't
+        // match - fill with zeros
+        for (auto &stress : stresses)
+          stress = 0.0;
+      }
+    else
+      {
+        const ArrayView<const Tensor<1, spacedim>>
+          cell_fibers = fiber_network->get_fibers(cell);  // cell specific fiber fields
+        for (unsigned int qp_n = 0; qp_n < stresses.size(); ++qp_n)
+          {
+            // stress contribution, isotropic term
+            stresses[qp_n] =
+              0.5 * a * std::exp(b * (m_values.get_modified_first_invariant()[qp_n] - 3.0)) * 
+              m_values.get_modified_first_invariant_dFF()[qp_n];
+            // stress contribution, transversly isotropic term, fiber f
+            const double I4_f = 
+              (m_values.get_right_cauchy_green()[qp_n] * cell_fibers[index_f]) * cell_fibers[index_f];
```

Comment:
Could we add (like in the heart model) helper functions for these higher up in this file? We'll probably want to reuse them for other fiber-reinforced models.

Reply:
sure, I can add some `inline` functions for these

Reply:
@drwells Where do you want the functions declared? Like in `ModifiedHolzapfelOgdenStress`?

Reply:
The header is a good choice - that way we can reuse this stuff in the heart model.